### ISSUE TITLE
SC2086 Rule violation was not reported in all instances occasionally.

### ISSUE THREAD

**Issue by ye** [koalaman/shellcheck#889]
#### For bugs
- Rule Id: SC2086
- My shellcheck version: 0.4.6
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [ ] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#! /bin/bash

# error out if any undefined variables are used
set -u

BASE_PATH="etl_data"
echo "BASE_PATH is ${BASE_PATH}"

# This is needed because it is run under cron (which empties out the ENV)
echo "PATH is: "
# shellcheck disable=SC2086
echo $PATH
PATH=/usr/local/bin:/usr/bin:/bin
echo "PATH is (after update): "
echo $PATH

# TODO: this is not atomic - files could enter the bucket during the cp time and not be copied yet
# copy usage logs to processing bucket
/events_etl/google-cloud-sdk/bin/gsutil cp gs://${LOGS_BUCKET_NAME}/${PIXEL_BUCKET_NAME}_usage* /${BASE_PATH}/

```

#### Here's what shellcheck currently says:

![screen shot 2017-04-25 at 11 55 16 am](https://cloud.githubusercontent.com/assets/133924/25395124/0e4b1030-29ae-11e7-838c-972e8e805e78.png)

$ shellcheck events_etl/etl.sh 

```
In events_etl/etl.sh line 19:
/events_etl/google-cloud-sdk/bin/gsutil cp gs://${LOGS_BUCKET_NAME}/${PIXEL_BUCKET_NAME}_usage* /${BASE_PATH}/
                                                ^-- SC2086: Double quote to prevent globbing and word splitting.
                                                                    ^-- SC2086: Double quote to prevent globbing and word splitting.
```

#### Here's what I wanted or expected to see:

Basically it misses the the last `${BASE_PATH}` for rule SC2086.

```
$ shellcheck events_etl/etl.sh 

In events_etl/etl.sh line 37:
/events_etl/google-cloud-sdk/bin/gsutil cp gs://${LOGS_BUCKET_NAME}/${PIXEL_BUCKET_NAME}_usage* /${BASE_PATH}/
                                                ^-- SC2086: Double quote to prevent globbing and word splitting.
                                                                    ^-- SC2086: Double quote to prevent globbing and word splitting.
                                                                                                 ^-- SC2086: Double quote to prevent globbing and word splitting.
```

What's even more interesting is that if I single out the offending line,  copy-paste it into a standalone shell script file, shellcheck is able to report it correctly. So what am I missing here?

```
$ cat test.sh
#!/bin/bash

/events_etl/google-cloud-sdk/bin/gsutil cp gs://${LOGS_BUCKET_NAME}/${PIXEL_BUCKET_NAME}_usage* /${BASE_PATH}/
$ shellcheck test.sh 

In test.sh line 3:
/events_etl/google-cloud-sdk/bin/gsutil cp gs://${LOGS_BUCKET_NAME}/${PIXEL_BUCKET_NAME}_usage* /${BASE_PATH}/
                                                ^-- SC2086: Double quote to prevent globbing and word splitting.
                                                                    ^-- SC2086: Double quote to prevent globbing and word splitting.
                                                                                                 ^-- SC2086: Double quote to prevent globbing and word splitting.
```

**Comment by koalaman** [comment]
ShellCheck tries (and sometimes fails) to be clever about this. 

Since `BASE_PATH` is a constant without metacharacters, quoting doesn't actually matter, so no 2086 is reported.

There should really be either a verbosity option or a separate style/info suggestion though.

**Comment by koalaman** [comment]
Optional warning available in https://github.com/koalaman/shellcheck/wiki/SC2248
